ibis/pandas/execution.py

```diff
@@ -270,30 +272,51 @@ def execute_cast_string_literal(op, data, type, scope=None):
     pd.Series,
     (pd.Series, np.integer, type(None)) + six.integer_types
 )
-def execute_round_series(op, data, places, scope=None):
+def execute_round_series(op, data, places, **kwargs):
     if data.dtype == np.dtype(np.object_):
-        return vectorize_object(op, data, places, scope=scope)
+        return vectorize_object(op, data, places, **kwargs)
     return data.round(places if places is not None else 0)
 
 
 @execute_node.register(ops.TableColumn, (pd.DataFrame, DataFrameGroupBy))
-def execute_table_column_dataframe_or_dataframe_groupby(op, data, scope=None):
+def execute_table_column_df_or_df_groupby(op, data, **kwargs):
     return data[op.name]
 
 
-def _compute_sort_key(key, scope):
+def _compute_sort_key(key, data, **kwargs):
     by = key.args[0]
     try:
         return by.get_name(), None
     except com.ExpressionError:
         name = util.guid()
-        new_column = execute(by, scope)
+        new_scope = {t: data for t in by.op().root_tables()}
+        new_column = execute(by, new_scope, **kwargs)
         new_column.name = name
         return name, new_column
 
 
+def _compute_sorted_frame(sort_keys, df, **kwargs):
+    computed_sort_keys = []
+    ascending = [key.op().ascending for key in sort_keys]
+    new_columns = {}
+
+    for i, key in enumerate(map(operator.methodcaller('op'), sort_keys)):
+        computed_sort_key, temporary_column = _compute_sort_key(
+            key, df, **kwargs
+        )
+        computed_sort_keys.append(computed_sort_key)
+
+        if temporary_column is not None:
+            new_columns[computed_sort_key] = temporary_column
+
+    result = df.assign(**new_columns)
+    result = result.sort_values(computed_sort_keys, ascending=ascending)
+    result = result.drop(new_columns.keys(), axis=1)
+    return result
+
+
 @execute_node.register(ops.Selection, pd.DataFrame)
-def execute_selection_dataframe(op, data, scope=None):
+def execute_selection_dataframe(op, data, scope=None, **kwargs):
```

Comment:
I need to refactor this function a bit after this PR is merged. It's getting a bit too big for its britches.